### ISSUE TITLE
Application crashes on a physical device

### ISSUE THREAD

**Issue by daumantas-versockas** [tuist/tuist#5430]
### What happened?

After migration to Tuist, the application crashes while running on a physical device (I have used Tuist to generate build configurations).

The application does not crash when I am not using build configurations but using a default .plist instead.

### How do we reproduce it?

1. Generate a [sample project](https://github.com/tuist/tuist/files/12665822/tuist-crash-demo.zip) using Tuist.
2. Run project on a physical device.
3. Inspect the application crash.

### Error log

```sh
dyld[77420]: Library not loaded: /Library/Frameworks/FeatureAImplementation.framework/FeatureAImplementation
  Referenced from: <53403544-CE9E-3521-AE9B-8E5208C834DC> /private/var/containers/Bundle/Application/793F6DA5-D5D8-4DCD-80C9-6FC69BE04930/Crasher.app/Crasher
  Reason: tried: '/Library/Frameworks/FeatureAImplementation.framework/FeatureAImplementation' (no such file), '/private/preboot/Cryptexes/OS/Library/Frameworks/FeatureAImplementation.framework/FeatureAImplementation' (no such file), '/Library/Frameworks/FeatureAImplementation.framework/FeatureAImplementation' (no such file), '/System/Library/Frameworks/FeatureAImplementation.framework/FeatureAImplementation' (no such file, not in dyld cache)
```

### macOS version

13.5.2 (22G91)

### Tuist version

3.23.1

### Xcode version

14.3.1

**Comment by kaisegni** [comment]
Hey, Have you managed to solve the issue? running into the same problem here

**Comment by pepicrft** [comment]
Thanks for reporting this and including the reproducible example @daumantas-versockas. We'll try to look at it ASAP.

**Comment by danibachar** [comment]
I wanted to share that I run into this issue with higher versions of tuist.

I could not put my finger on exactly what it is (combed through the configurations), but its some configuration in the XCConfig file. Even with a single framework linked to the app (for example `ImplementationA` and `InterfaceA` you can reproduce the issue.

Removing those make the app build and run on device, I tested with tuist 4.22.0 and 4.26.0 - on both its reproducible with the Project.xcconfig files and solved when removing them.

I'm facing a similar issue with our main app

**Comment by kaisegni** [comment]
I managed to find a solution for my issue, and it is essentially adding the library/ dependency to the main target so if you have App as main target, and a Networking module and you use library x  as part of Networking then adding `.package(product: "x"),`  to both App Target as well as Networking target solved it for me. I hope it helps

**Comment by fortmarek** [comment]
Hey 👋 

I looked into the provided sample and when I set the `DYLIB_BASE_NAME_BASE` to `@rpath` for the `Crasher` and `FeatureAImplementation` targets – which is the value we set in the [essential setting](https://github.com/tuist/XcodeProj/blob/46fe15644abd14df0c288ba55ab00eda32e78a18/Sources/XcodeProj/Utils/BuildSettingsProvider.swift#L214) – the app no longer crashes.

I'd suggest using at least `.essential` unless you _really_ need the extra flexibility. Even better if you use `.recommended` as your default settings.

I'm gonna close this issue as there's nothing we can do in tuist directly to fix this as we don't tinker with the settings at all if you choose to use `.none` for your settings.

If there are still some issues with the fixture, we can reopen the issue if needed.
